src/crimson/osd/osd.cc

```diff
@@ -542,13 +542,15 @@ seastar::future<Ref<PG>> OSD::make_pg(cached_map_t create_map,
 
 seastar::future<Ref<PG>> OSD::load_pg(spg_t pgid)
 {
-  return PGMeta{store.get(), pgid}.get_epoch().then([this](epoch_t e) {
-    return get_map(e);
-  }).then([pgid, this] (auto&& create_map) {
+  return seastar::do_with(PGMeta(store.get(), pgid), [this, pgid] (auto& pg_meta) {
```

Comment:
could you change the title of the commit message to something like

> crimson/osd: ensure pg_meta is alive when calling its methods

as "fix pg meta issue" is not specific enough.